test/1.4/misc/size_stats.dml

```diff
@@ -0,0 +1,99 @@
+/*
+  © 2021-2022 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+
+dml 1.4;
+
+device test;
+
+/// COMPILE-ONLY
+
+/*
+  This test creates some methods that generate a lot of C code, and see that
+  they show up when compiling with DMLC_GATHER_SIZE_STATISTICS.
+
+  First, the baseline function is a method body that generates about 10*64*20 =
+  12k bytes of code (-1 expands to a ~20 byte uint64 literal, and we output 10
+  variables with 64 * -1 as initializers).
+
+  Then two typical code explosion patterns are exercised: one is to instantiate
+  many non-shared implementations of a method, and one is to expand a large
+  body using #foreach or #select.
```

Comment:
The test may look a bit weird. I wanted something that's fast to compile, yet generates C functions dominated by predictably sized parts, and whose size exceeds that of the largest dml-builtins method by a good margin (2.5x). We may have to adjust sizes someday but it shouldn't need to happen often.